playbooks/roles/filebeat/templates/filebeat.yml.j2

```diff
@@ -0,0 +1,11 @@
+filebeat:
+  registry_file: "{{ filebeat_registry_file }}"
+
+  prospectors:
+    {% for prospector in filebeat_prospectors %}
+- {{ filebeat_common_prospector | combine(prospector, recursive=True) | to_nice_yaml | indent(width=6) }}
+    {% endfor %}
```

Comment:
I can see why it is behaving the way it behaves, but it's really not obvious.  This is a result of Jinja2's whitespace stripping behviour, which usuall behaves as intended, but can be difficult to exactly predict.  That's the reason why I suggested using `to_json` here – it will format the dictionary without any newline characters, so you don't run into the any problems with indentation.

For what it's worth, here is some code you could include in `defaults/main.yml` as an alternative:

    filebeat_prospectors_yaml: |
      {% for prospector in filebeat_prospectors %}
      - {{ filebeat_common_prospector | combine(prospector, recursive=True) | to_json }}
      {% endfor %}
    filebeat_config:
      filebeat:
        registry_file: "{{ filebeat_registry_file }}"
        prospectors: "{{ filebeat_prospectors_yaml | from_yaml }}"
      output:
        logstash: "{{ filebeat_logstash }}"

With these definitions, writing `filebeat.yml` becomes a matter of `{{ filebeat_config | to_nice_yaml }}`, and we don't need any fragile indentation code.  (I wrote this code just out of curiosity – completely up to you whether you want to include it.  It's still messy – we take a list of dictionaries, format them as JSON, combine them into a YAML-formatted string, then convert back to YAML just to format as YAML again.  The last two steps are for a nice-looking output.)

Reply:
Yeah at this point I'll probably just leave this alone since I don't have more time to try other ways of doing this. I'd also like to keep the Jinja2 code inside of `j2` files to make it easier to understand what's going on. I think as long as one understands how `indent` is working in the current setup, things should be okay.